### ISSUE TITLE
Build failure with clang 12 in common/http/HTTPServer.cpp on macOS

### ISSUE THREAD

**Issue by fxcoudert** [OpenLightingProject/ola#1672]
Compiling ola-0.10.7 with clang 12 on macOS (Xcode 12 system compiler) leads to this build failure:

```
 common/http/HTTPServer.cpp:221:3: error: no matching function for call to 'MHD_get_connection_values'
   MHD_get_connection_values(m_connection, MHD_HEADER_KIND, AddHeaders, this);
   ^~~~~~~~~~~~~~~~~~~~~~~~~
 /usr/local/include/microhttpd.h:2697:1: note: candidate function not viable: no known conversion from 'int (void *, enum MHD_ValueKind, const char *, const char *)' to 'MHD_KeyValueIterator' (aka 'enum MHD_Result (*)(void *, enum MHD_ValueKind, const char *, const char *)') for 3rd argument
 MHD_get_connection_values (struct MHD_Connection *connection,
 ^
 common/http/HTTPServer.cpp:224:19: error: no matching function for call to 'MHD_create_post_processor'
     m_processor = MHD_create_post_processor(m_connection,
                   ^~~~~~~~~~~~~~~~~~~~~~~~~
 /usr/local/include/microhttpd.h:3463:1: note: candidate function not viable: no known conversion from 'int (void *, enum MHD_ValueKind, const char *, const char *, const char *, const char *, const char *, uint64_t, size_t)' (aka 'int (void *, MHD_ValueKind, const char *, const char *, const char *, const char *, const char *, unsigned long long, unsigned long)') to 'MHD_PostDataIterator' (aka 'MHD_Result (*)(void *, MHD_ValueKind, const char *, const char *, const char *, const char *, const char *, unsigned long long, unsigned long)') for 3rd argument
 MHD_create_post_processor (struct MHD_Connection *connection,
 ^
 common/http/HTTPServer.cpp:468:13: error: no matching function for call to 'MHD_start_daemon'
   m_httpd = MHD_start_daemon(MHD_NO_FLAG,
             ^~~~~~~~~~~~~~~~
 /usr/local/include/microhttpd.h:2425:1: note: candidate function not viable: no known conversion from 'int (*)(void *, struct MHD_Connection *, const char *, const char *, const char *, const char *, size_t *, void **)' (aka 'int (*)(void *, MHD_Connection *, const char *, const char *, const char *, const char *, unsigned long *, void **)') to 'MHD_AccessHandlerCallback' (aka 'MHD_Result (*)(void *, MHD_Connection *, const char *, const char *, const char *, const char *, unsigned long *, void **)') for 5th argument
 MHD_start_daemon (unsigned int flags,
 ^
```

The exact same build with previous versions of the compiler passed, probably because the compiler was not that strict with checking prototypes.

**Comment by peternewman** [comment]
Hi @fxcoudert ,

This is actually due to a breaking change in libmicrohttpd . I'm sort of surprised we're the only people impacted.

This bug is a duplicate of #1650 which has been fixed in #1651 .

We've not released yet, so for now you'll need to take a patch from common/http/HTTPServer.cpp and include/ola/http/HTTPServer.h in that PR.

Or roll libmicrohttpd back to a version prior to v0.9.71 .

**Comment by peternewman** [comment]
Closing this now as it's a duplicate, merged pending release, patched in Homebrew and shortly in Macports ( https://github.com/macports/macports-ports/pull/8941 ).
